### PR TITLE
feat: add group to role rebac admin handlers

### PR DESCRIPTION
**Pull request by kian99**
## Description

This PR implements the `GetGroupRoles` and `PatchGroupRoles` allowing groups to be assigned roles.

Fixes and completes [JUJU-7127](https://warthogs.atlassian.net/browse/JUJU-7127)

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [x] Covered by unit tests
- [x] Covered by integration tests

[JUJU-7127]: https://warthogs.atlassian.net/browse/JUJU-7127?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SimoneDutto — APPROVED** [review]
lgtm

File: internal/jimmhttp/rebac_admin/groups.go [hunk 1]
```diff
@@ -236,12 +237,116 @@ func (s *groupsService) PatchGroupIdentities(ctx context.Context, groupId string
...
+		resp.Next = resources.Next{
+			PageToken: &nextToken,
+		}
+	}
+	return &resp, nil
 }
 
-// PatchGroupRoles performs addition or removal of a Role to/from a Group identified by `groupId`.
+// PatchGroupRoles performs addition or removal of a group to/from a role identified by `groupId`.
 func (s *groupsService) PatchGroupRoles(ctx context.Context, groupId string, rolePatches []resources.GroupRolesPatchItem) (bool, error) {
```
Comment (ale8k) [inline internal/jimmhttp/rebac_admin/groups.go:304]: What is the purpose of this func?...

**Review by ale8k — APPROVED** [review]
lgtm just confused why we need patching

File: internal/jimmhttp/rebac_admin/groups.go (hunk 1 above)
Comment (kian99) [reply internal/jimmhttp/rebac_admin/groups.go:304]: It just assigns/removes groups from roles.

Comment (kian99) [reply internal/jimmhttp/rebac_admin/groups.go:304]: Or better phrased, assigns roles to groups or un-assigns roles from groups.

**Pull request merged**